src/code/z_player_lib.c

```diff
@@ -1611,68 +1681,172 @@ void Player_DrawImpl(PlayState* play, void** skeleton, Vec3s* jointTable, s32 dL
     POLY_OPA_DISP = &gfx[2];
 
     D_801F59E0 = playerForm * 2;
-    D_801F59E4 = lod;
+    sPlayerLod = lod;
     SkelAnime_DrawFlexLod(play, skeleton, jointTable, dListCount, overrideLimbDraw, postLimbDraw, actor, lod);
 
     CLOSE_DISPS(play->state.gfxCtx);
 }
 
 Vec3f D_801C08C0[PLAYER_FORM_MAX] = {
-    { 1304.0f, 0.0f, 0.0f }, { 1156.0f, 0.0f, 0.0f }, { 1406.0f, 0.0f, 0.0f },
-    { 408.0f, 0.0f, 0.0f },  { 695.0f, 0.0f, 0.0f },
+    { 1304.0f, 0.0f, 0.0f }, // PLAYER_FORM_FIERCE_DEITY
+    { 1156.0f, 0.0f, 0.0f }, // PLAYER_FORM_GORON
+    { 1406.0f, 0.0f, 0.0f }, // PLAYER_FORM_ZORA
+    { 408.0f, 0.0f, 0.0f },  // PLAYER_FORM_DEKU
+    { 695.0f, 0.0f, 0.0f },  // PLAYER_FORM_HUMAN
 };
 
 f32 D_801C08FC[PLAYER_FORM_MAX] = {
-    1265.0f, 1056.0f, 1506.0f, 359.0f, 826.0f,
+    1265.0f, // PLAYER_FORM_FIERCE_DEITY
+    1056.0f, // PLAYER_FORM_GORON
+    1506.0f, // PLAYER_FORM_ZORA
+    359.0f,  // PLAYER_FORM_DEKU
+    826.0f,  // PLAYER_FORM_HUMAN
 };
 f32 D_801C0910[PLAYER_FORM_MAX] = {
-    170.0416f, 133.63359f, 197.68358f, 16.646399f, 48.302498f,
+    170.0416f,  // PLAYER_FORM_FIERCE_DEITY
+    133.63359f, // PLAYER_FORM_GORON
+    197.68358f, // PLAYER_FORM_ZORA
+    16.646399f, // PLAYER_FORM_DEKU
+    48.302498f, // PLAYER_FORM_HUMAN
 };
 f32 D_801C0924[PLAYER_FORM_MAX] = {
-    10.019104f, 22.120003f, -29.12001f, 3.7582989f, -19.925102f,
+    10.019104f,  // PLAYER_FORM_FIERCE_DEITY
+    22.120003f,  // PLAYER_FORM_GORON
+    -29.12001f,  // PLAYER_FORM_ZORA
+    3.7582989f,  // PLAYER_FORM_DEKU
+    -19.925102f, // PLAYER_FORM_HUMAN
 };
 f32 D_801C0938[PLAYER_FORM_MAX] = {
-    5.0f, 4.0f, 1.0f, 1.0f, 3.0f,
+    5.0f, // PLAYER_FORM_FIERCE_DEITY
+    4.0f, // PLAYER_FORM_GORON
+    1.0f, // PLAYER_FORM_ZORA
+    1.0f, // PLAYER_FORM_DEKU
+    3.0f, // PLAYER_FORM_HUMAN
 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124870.s")
+/**
+ * Adjuts player's legs to slopes and spawns flames when running on lava
+ */
+void func_80124870(PlayState* play, Player* player, SkelAnime* skelAnime, Vec3f* pos, Vec3s* rot, s32 thighLimbIndex,
+                   s32 shinLimbIndex, s32 footLimbIndex) {
+    CollisionPoly* poly;
+    s32 bgId;
+    f32 yIntersect;
+    Vec3f sp90;
+    Vec3f sp84;
+    Vec3f footprintPos;
+    f32 sp74;
+    f32 sp70;
+    f32 sp7C;
+
+    if ((player->stateFlags3 & PLAYER_STATE3_1) || !(player->actor.scale.y >= 0.0f) ||
+        (player->stateFlags1 & PLAYER_STATE1_80) || (play->unk_18844 != 0)) {
```

Comment:
```suggestion
        (player->stateFlags1 & PLAYER_STATE1_80) || play->unk_18844) {
```
`unk_18844` looks boolean